### PR TITLE
Update remote access

### PR DESCRIPTION
**Pull request by Sigabort42**
add a method update_remote_access

### CODE REVIEW

File: cbw_api_toolbox/cbw_api.py [hunk 1]
```diff
@@ -106,3 +106,21 @@ def remote_accesses(self):
         """GET request to /api/v2/remote_accesses to get all servers"""
         response = self._request("GET", [ROUTE_REMOTE_ACCESSES])
         return CBWParser().parse_response(CBWRemoteAccess, response)
+
+    def update_remote_access(self, remote_id, info):
+        """PATCH request to /api/v2/remote_accesses to update a specific server"""
```
Comment (jinxka) [inline cbw_api_toolbox/cbw_api.py]: PATCH request to /api/v2/remote_accesses/{remote_id} to update a remote access

File: spec/test_cbw_api.py [hunk 2]
```diff
@@ -2,7 +2,6 @@
 
 from cbw_api_toolbox.cbw_api import CBWApi
 from cbw_api_toolbox.cbw_objects.cbw_server import CBWServer
-from cbw_api_toolbox.cbw_objects.cbw_remote_access import CBWRemoteAccess
```
Comment (jinxka) [inline spec/test_cbw_api.py]: why?

File: spec/test_cbw_api.py [hunk 3]
```diff
@@ -99,12 +98,48 @@ def test_update_server():
             assert response is True
 
     @staticmethod
-    def test_remote_accesses():
```
Comment (jinxka) [inline spec/test_cbw_api.py]: why?

File: spec/test_cbw_api.py [hunk 4]
```diff
@@ -99,12 +98,48 @@ def test_update_server():
...
+                "key": "",  # precises the key of the connection
+                "node": "master"  # precises the Cyberwatch source of the connection,
+                }
+
+        with vcr.use_cassette('spec/fixtures/vcr_cassettes/update_remote_access_id_none.yaml'):
+            response = client.update_remote_access(None, info)
+
+            assert response is False
+
+        info = {"type": "",
```
Comment (jinxka) [inline spec/test_cbw_api.py]: if the type is the only field changing, you can just write info["type"] = ""

File: spec/test_cbw_api.py [hunk 5]
```diff
@@ -99,12 +98,48 @@ def test_update_server():
...
+                "key": "",
+                "node": "master"
+                }
+
+        with vcr.use_cassette('spec/fixtures/vcr_cassettes/update_remote_access.yaml'):
+            response = client.update_remote_access('4', info)
+
+            assert response is True
+
+        info = {"type": "CbwRam::RemoteAccess::Ssh::WithPassword",
```
Comment (jinxka) [inline spec/test_cbw_api.py]: there are no difference with the previous one, remove.

**Comment by jinxka** [comment]
The example file is missing

File: cbw_api_toolbox/cbw_api.py (hunk 1 above)
Comment (Sigabort42) [reply cbw_api_toolbox/cbw_api.py]: :)

File: spec/test_cbw_api.py (hunk 2 above)
Comment (Sigabort42) [reply spec/test_cbw_api.py]: Because it's not used by the update_remote_access test

File: spec/test_cbw_api.py (hunk 5 above)
Comment (Sigabort42) [reply spec/test_cbw_api.py]: there is one in the address

Comment (Sigabort42) [reply spec/test_cbw_api.py]: Ok :-1:

File: spec/test_cbw_api.py (hunk 4 above)
Comment (Sigabort42) [reply spec/test_cbw_api.py]: Ok :+1:

File: examples/update_remote_access.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+"""Update remote access agentless"""
```
Comment (fwininger) [inline examples/update_remote_access.py]: Remove agentless

File: examples/update_remote_access.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+        "port": "", #mandatory, precises the port of the connection
+        "login": "", #mandatory, precises the login of the connection
+        "password": "", #precises the password of the connection
+        "key": "", #precises the key of the connection
+        "node": "" #precises the Cyberwatch source of the connection
+        }
+
+REMOTE_ID = ''
+
+CLIENT.update_remote_access(REMOTE_ID, INFO)
```
Comment (fwininger) [inline examples/update_remote_access.py]: No new line

File: examples/update_remote_access.py [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+INFO = {"type": "", #mandatory, precises the type of the connection
+        "address": "", #mandatory, precises the IP address or the domain name of the targeted computer
+        "port": "", #mandatory, precises the port of the connection
+        "login": "", #mandatory, precises the login of the connection
+        "password": "", #precises the password of the connection
+        "key": "", #precises the key of the connection
+        "node": "" #precises the Cyberwatch source of the connection
+        }
+
+REMOTE_ID = ''
```
Comment (fwininger) [inline examples/update_remote_access.py]: REMOTE_ACCESS_ID

File: cbw_api_toolbox/cbw_api.py [hunk 9]
```diff
@@ -139,3 +139,21 @@ def remote_access(self, remote_access_id):
             return None
 
         return CBWParser().parse_response(CBWRemoteAccess, response)
+
+    def update_remote_access(self, remote_id, info):
```
Comment (fwininger) [inline cbw_api_toolbox/cbw_api.py]: remote_access_id

**Pull request merged**